workflow/airflow/dags/local_file_sensor_dag.py

```diff
@@ -0,0 +1,94 @@
+import os
+
+import airflow
+from airflow import DAG
+from airflow.operators.python_operator import PythonOperator
+from airflow.operators.sensors import BaseSensorOperator
+from datetime import timedelta
+
+
+class FileWatcherOperator(BaseSensorOperator):
+    """
+    Triggers a file system check. Parses a file if present.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize a FileWatcherOperator which should wait for a file to be present in a target directory.
+
+        :param args:
+        :param kwargs:
+        """
+        self.directory_path = kwargs.get('directory_path')
+        super(FileWatcherOperator, self).__init__(*args, **kwargs)
+
+    def poke(self, context, *args, **kwargs):
+        """
+        See if a file is present in a given directory.
+
+        :param context: Context dictionary forwarded from the DAG;
+        expects to contain context.dag and context.task
+        :type context: dict
+        :param args: not expected
+        :param kwargs: not expected
+        :return: whether the poke method has succeeded or not, 0 or 1
+        :rtype: int
+        """
+        path = os.path.join(os.getcwd(), self.directory_path)
+        dir_listing = os.listdir(path)
+
+        if len(dir_listing) > 0:
+            context['ti'].xcom_push(key='file_path', value=os.path.join(path, dir_listing[0]))
```

Comment:
so this is how operators communicate. Is `file_path` available only in the directly downstream operator, or does it become more "global" in further operators down the workflow?

Reply:
Yes their XCOM system is how you can communicate between steps in a workflow. It's effectively a key-value store in which a workflow instance will have a namespace.

You can access the `return_value` of a given operation by just referring to a `task_id` or you can push and pull via a `key` which I found more flexible and clearer.